lib/faraday_middleware/response/parse_yaml.rb

```diff
@@ -1,29 +1,40 @@
+require 'psych'
 require 'faraday_middleware/response_middleware'
 
 module FaradayMiddleware
   # Public: Parse response bodies as YAML.
   #
-  # Warning: this uses `YAML.load()` by default and as such is not safe against
-  # code injection or DoS attacks. If you're loading resources from an
-  # untrusted host or over HTTP, you should subclass this middleware and
-  # redefine it to use `safe_load()` if you're using a Psych version that
-  # supports it:
+  # Warning: This is not backwards compatible with versions of this middleware prior to
+  # faraday_middleware v0.12 - prior to this version, we used YAML.load rather than
+  # YAMl.safe_load, which exposes serious remote code execution risks - see
+  # https://github.com/ruby/psych/issues/119 for details. If you're sure you can trust
+  # YAML you're passing, you can set up an unsafe version of the middleware as follows:
   #
-  #     class SafeYaml < FaradayMiddleware::ParseYaml
+  #     class UnsafeYaml < FaradayMiddleware::ParseYaml
   #       define_parser do |body|
-  #         YAML.safe_load(body)
+  #         Psych.load body
   #       end
   #     end
   #
   #     Faraday.new(..) do |config|
-  #       config.use SafeYaml
+  #       config.use UnsafeYaml
   #       ...
   #     end
   class ParseYaml < ResponseMiddleware
-    dependency 'yaml'
+    dependency do
+      if RUBY_VERSION =~ /^2./
+        require 'psych'
+      else
+        require 'safe_yaml/load'
+      end
+    end
 
     define_parser do |body|
-      ::YAML.load body
+      if RUBY_VERSION =~ /^2./
+        Psych.safe_load body
+      else
+        SafeYAML.load body
+      end
```

Comment:
I'm a bit short on time so can you please expose the difference between one and the other?
How would this code change, assuming we might drop support for some ruby versions?